lib/api/shared/src/main/scala/net/wiringbits/api/ApiClient.scala

```diff
@@ -19,6 +20,14 @@ class ApiClient(config: ApiClient.Config)(implicit
     ex: ExecutionContext,
     sttpBackend: SttpBackend[Future, _]
 ) {
+  // Instead of using a random UUID we can avoid generating a random UUID mocking a valid UUID,
+  // this UUID is not being used in the server/api client, so it is safe to use it
+  private val mockedUserId = Future.successful(UUID.fromString("887a5d77-cb5d-4d9c-b4dc-539c8aae3977"))
+
+  // We need a function to handle endpoints that need user authentication, this is not being used in the server/api
+  // client, so it is safe to use it
+  private implicit val handleUserId: ServerRequest => Future[UUID] = _ => mockedUserId
```

Comment:
I'd say `Similarly to the dummy userId, we need a way to derive the userId from a request, which is used only on the server-side code, this function is helpful to fulfill the method signatures`